test/files/neg/checksensible.check

```diff
@@ -34,12 +34,30 @@ checksensible.scala:31: warning: comparing values of types Int and String using
 checksensible.scala:35: warning: comparing values of types Some[Int] and Int using `==` will always yield false
   Some(1) == 1      // as above
           ^
+#partest !java8
+checksensible.scala:37: warning: constructor Boolean in class Boolean is deprecated
+  true == new java.lang.Boolean(true) // none of these should warn except for deprecated API
```

Comment:
Could we just use `java.lang.Boolean.valueOf` in the test to avoid the conditional sections in the checkfile?

Reply:
Slapping `@deprecated` on the class suffices to silence the warnings. In tests, either the code is not invoked at all, or it's invoked reflectively.

I suspect it's trying to provoke one of the heuristics with `new`, but I didn't look to refresh my vague memory.

Reply:
Somewhere I commented that I'd like to try removing the inline filter with separate check files for different platform versions. I bet other check files have been silently overwritten with homogeneous output.

Reply:
Hehehehe, the situation is quite ironic:

* bf4c304a71672d375e5f7d238e679b1826cc7489 Som sets up pragmas
* b0192f748626691d8252b756d92ded05eca78089 Som overwrites!
* 3e7e0ab671f2d83ab09940444027214bcc346cc6 Lukas restores
* ec80eb49f99c3870f5de6d45e0f10a6947eafe9c Eugene overwrites

and now I restore.  Separate checkfiles could help, but also if these were run on JDK 11 in PR validation it would catch the mistake.

Reply:
It's only software. https://github.com/scala/scala/pull/9362